checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -201,6 +209,86 @@ private void checkOwningOverrides(
     }
   }
 
+  /**
+   * If a {@code @MustCallAlias} annotation appears in a method declaration, it must appear as an
+   * annotation on both the return type, and a parameter type.
+   *
+   * <p>The return type is checked if it is annotated with {@code @PolyMustCall} because the Must
+   * Call Checker treats {@code @MustCallAlias} as an alias of {@code @PolyMustCall}.
+   *
+   * @param tree the method declaration.
+   * @param mcAtf the MustCallAnnotatedTypeFactory.
+   */
+  private void checkMustCallAliasAnnotationForMethod(
+      MethodTree tree, MustCallAnnotatedTypeFactory mcAtf) {
+
+    boolean isMustCallAliasAnnoOnParameter = isMustCallAliasAnnoPresentInParams(tree, mcAtf);
+
+    if (TreeUtils.isVoidReturn(tree) && isMustCallAliasAnnoOnParameter) {
+      checker.reportWarning(tree, "mustcallalias.method.return.and.param");
+      return;
+    }
+
+    AnnotatedTypeMirror returnType = mcAtf.getMethodReturnType(tree);
+    boolean isMustCallAliasAnnoOnReturnType = returnType.hasPrimaryAnnotation(PolyMustCall.class);
+
+    if (isMustCallAliasAnnoOnParameter != isMustCallAliasAnnoOnReturnType) {
+      checker.reportWarning(tree, "mustcallalias.method.return.and.param");
```

Comment:
If you take my suggestion about the warnings, this is where you'll need to make a corresponding change.